apple/internal/aspects/docc_bundle_info_aspect.bzl

```diff
@@ -0,0 +1,92 @@
+"""Defines an aspect for collecting the DocCBundleInfo provider for targets that support it."""
+
+load(
+    "@build_bazel_rules_swift//swift/internal:providers.bzl",
+    "SwiftInfo",
+)
+load(
+    "//apple:providers.bzl",
+    "DocCBundleInfo",
+)
+
+_DOCC_BUNDLE_INFO_ASPECT_ATTRS = ["data", "deps"]
+
+def _top_level_swift_info(target, ctx):
```

Comment:
Doing some more testing:

- [ ] Need to be specific about which targets we allow this traversal for, we only want this to happen for packaging targets like `ios_framework`

Reply:
TIL more about aspects, I split this up into two aspects and two providers. One that gathers the symbol graphs and another the bundle information.

This matches how docc is used where you can generate for purely code, just a .docc bundle or both.